1-Vanilla/index.js

```diff
@@ -0,0 +1,170 @@
+            // item title
+            let titleElement = document.createElement('div')
+            titleElement.innerText = title
+            titleElement.className = 'todo-item-title'
+            titleElement.id = 'title'+id
+            // item subtitle
+            let subtitleElement = document.createElement('div')
+            subtitleElement.innerText = subtitle
+            subtitleElement.className = 'todo-item-subtitle'
+            subtitleElement.id = 'subtitle'+id
+            // item description
+            let descriptionElement = document.createElement('div')
+            descriptionElement.innerText = description
+            descriptionElement.className = 'todo-item-description'
+            descriptionElement.id = 'description'+id
+            // div todo-item-buttons
+        let itemButtonsElement = document.createElement('div')
+        itemButtonsElement.className = 'todo-item-buttons'
+            // button edit
+            let editButtonElement = document.createElement('button')
+            editButtonElement.innerText = 'edit'
+            editButtonElement.className = 'item-edit-button'
+            editButtonElement.setAttribute('onclick',`edit_item(${id})`)
+            // button delete
+            let deleteButtonElement = document.createElement('button')
+            deleteButtonElement.innerText = 'delete'
+            deleteButtonElement.className = 'item-delete-button'
+            deleteButtonElement.setAttribute('onclick',`delete_item(${id})`)
+
+    // append the text to parent
+    itemTextElement.appendChild(titleElement)
+    itemTextElement.appendChild(subtitleElement)
+    itemTextElement.appendChild(descriptionElement)
+    // append the buttons to parent
+    itemButtonsElement.appendChild(editButtonElement)
+    itemButtonsElement.appendChild(deleteButtonElement)
+    //append checkbox, text and buttons to list-item
+    itemElement.appendChild(itemCheckboxElement)
+    itemElement.appendChild(itemTextElement)
+    itemElement.appendChild(itemButtonsElement)
+
+    // the itemElement should look like this example:
+
+    // <ol id='item1' class='todo-item'>
+    //    <input class='item-checkbox' type='checkbox'/>
+    //    <div class='todo-item-text'>
+    //        <div class='todo-item-title' id='title1'>Go for a walk with the dog</div>
+    //        <div class='todo-item-subtitle' id='subtitle1'>very fun</div>
+    //        <div class='todo-item-description' id='description1'>walk around the block 3 times, bring a doggy bag</div>
+    //    </div>
+    //    <div class='todo-item-buttons'>
+    //        <button class='item-edit-button' onclick='edit_item(1)'>edit</button>
+    //        <button class='item-delete-button' onclick='delete_item(1)'>delete</button>
+    //    </div> 
+    // </ol>
+
+    // finally append item to the list
+    list.appendChild(itemElement)
+
+    // reset menu display
+    document.getElementById('title-input').value = ''
```

Comment:
`document.getElementById(X).value` is something that you have a lot of.. think of a way to shorten it for example
setElementValue(x) or even just setValue(X)